modules/rubiconAnalyticsAdapter.js

```diff
@@ -0,0 +1,441 @@
+          cache.auctions[args.auctionId].bidsWon[bid.adUnitCode] = false;
+
+          memo[bid.bidId] = _pick(bid, [
+            'bidder', bidder => bidder.toLowerCase(),
+            'bidId',
+            'status', () => 'no-bid', // default a bid to no-bid until response is recieved or bid is timed out
+            'finalSource as source',
+            'params', (params, bid) => {
+              switch (bid.bidder) {
+                // specify bidder params we want here
+                case 'rubicon':
+                  return _pick(params, [
+                    'accountId',
+                    'siteId',
+                    'zoneId'
+                  ]);
+              }
+            },
+            'videoAdFormat', (_, cachedBid) => {
+              if (cachedBid.bidder === 'rubicon') {
+                return ({
+                  201: 'pre-roll',
+                  202: 'interstitial',
+                  203: 'outstream',
+                  204: 'mid-roll',
+                  205: 'post-roll',
+                  207: 'vertical'
+                })[utils.deepAccess(bid, 'params.video.size_id')];
+              } else {
+                let startdelay = parseInt(utils.deepAccess(bid, 'params.video.startdelay'), 10);
+                if (!isNaN(startdelay)) {
+                  if (startdelay > 0) {
+                    return 'mid-roll';
+                  }
+                  return ({
+                    '0': 'pre-roll',
+                    '-1': 'mid-roll',
+                    '-2': 'post-roll'
+                  })[startdelay]
+                }
+              }
+            },
+            'adUnit', () => _pick(bid, [
+              'adUnitCode',
+              'transactionId',
+              'sizes as dimensions', sizes => sizes.map(sizeToDimensions),
+              'mediaTypes', (types) => {
+                if (bid.mediaType && validMediaType(bid.mediaType)) {
+                  return [bid.mediaType];
+                }
+                if (Array.isArray(types)) {
+                  return types.filter(validMediaType);
+                }
+                if (typeof types === 'object') {
+                  if (!bid.sizes) {
+                    bid.dimensions = [];
+                    utils._each(types, (type) =>
+                      bid.dimensions = bid.dimensions.concat(
+                        type.sizes.map(sizeToDimensions)
+                      )
+                    );
+                  }
+                  return Object.keys(types).filter(validMediaType);
+                }
+                return ['banner'];
+              }
+            ])
+          ]);
+          return memo;
+        }, {}));
+        break;
+      case BID_RESPONSE:
+        let bid = cache.auctions[args.auctionId].bids[args.adId];
+        bid.source = formatSource(bid.source || args.source);
+        switch (args.getStatusCode()) {
+          case GOOD:
+            bid.status = 'success';
+            break;
+          case NO_BID:
```

Comment:
I think we do not have NO_BID case any more.
https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L299

Reply:
There's still some instances that can result in NO_BIDs such as in prebid server adapter, currency module, etc.  I'll probably leave this here just in case; I can remove later if NO_BIDs end up being removed entirely from the prebid code base.